[mono] Add an 'allow-loader-errors' AOT option.

vargaz opened the pull request:
This can be used to avoid aborting the AOT process if a loader
error occurs. The methods which fail to load will not be AOTed and
the failures will happen at runtime.

Related:
https://github.com/dotnet/runtime/issues/63654

lambdageek reviewed (commented):
it's not just loader errors.

There could be marshaler errors, too, for example: 
   https://github.com/dotnet/runtime/blob/53d52884198b15d83e1d9c2ce8009ef2c6a015c2/src/mono/mono/mini/aot-compiler.c#L5173-L5174

And there are other calls to `mono_error_assert_ok` in aot-compiler.c that could be masking loader errors.

vargaz committed:
[mono] Add an 'allow-errors' AOT option.

This can be used to avoid aborting the AOT process if a loader
error occurs. The methods which fail to load will not be AOTed and
the failures will happen at runtime.

Related:
https://github.com/dotnet/runtime/issues/63654

vargaz commented:
Reenabled the option to 'allow-errors'. If the tests hit one of those assertions, we can convert them to errors as well.
